kpm/lib/kpm/account.rb

```diff
@@ -0,0 +1,386 @@
+    KILLBILL_API_VERSION = '1.0'
+
+    # USER/PWD
+    KILLBILL_USER = 'admin'
+    KILLBILL_PASSWORD = 'password'
+
+    # TENANT KEY
+    KILLBILL_API_KEY = 'bob'
+    KILLBILL_API_SECRET = 'lazar'
+
+    # Temporary directory
+    TMP_DIR_PEFIX = 'killbill'
+    TMP_DIR = Dir.mktmpdir(TMP_DIR_PEFIX);
+
+    # Created By
+    WHO = 'kpm_export_import'
+
+    # safe payment method
+    SAFE_PAYMENT_METHOD = '__EXTERNAL_PAYMENT__'
+    PLUGIN_NAME_COLUMN = 'plugin_name'
+
+    # fields to remove from the export files
+    REMOVE_DATA_FROM = {:accounts => [:name, :address1, :address2, :city, :state_or_providence, :phone, :email],
+                       :account_history => [:name, :address1, :address2, :city, :state_or_providence, :phone, :email]}
+
+    def initialize(config_file = nil, killbill_api_credentials = nil, killbill_credentials = nil, killbill_url = nil, database = nil)
+      @killbill_api_key = KILLBILL_API_KEY
+      @killbill_api_secrets = KILLBILL_API_SECRET
+      @killbill_url = KILLBILL_URL
+      @killbill_user = KILLBILL_USER
+      @killbill_password = KILLBILL_PASSWORD
+
+      set_config(config_file)
+
+      if not @config.nil?
+        config_killbill = @config['killbill']
+
+        if not config_killbill.nil?
+          @killbill_api_key = config_killbill['api_key']
+          @killbill_api_secrets = config_killbill['api_secret']
+          @killbill_url = "http://#{config_killbill['host']}:#{config_killbill['port']}"
+          @killbill_user = config_killbill['user']
+          @killbill_password = config_killbill['password']
+        end
+
+        config_db = @config['database']
+
+        if not config_db.nil?
+          Database.set_configuration(config_db['database'], config_db['username'], config_db['password'])
+        end
+      end
+
+      if not killbill_api_credentials.nil?
+
+        @killbill_api_key = killbill_api_credentials[0]
+        @killbill_api_secrets = killbill_api_credentials[1]
+
+      end
+
+      if not killbill_credentials.nil?
+
+        @killbill_user = killbill_credentials[0]
+        @killbill_password = killbill_credentials[1]
+
+      end
+
+      if not killbill_url.nil?
+
+        @killbill_url = killbill_url
+
+      end
+
+      if not database.nil?
+
+        Database.set_configuration(database[0], database[1], database[2])
+
+      end
+    end
+
+    def set_config(config_file = nil)
+      @config = nil
+
+      if not config_file.nil?
+        if not Dir[config_file][0].nil?
+          @config = YAML::load_file(config_file)
+        end
+      end
+
+    end
+
+    def export_data(account_id = nil)
+
+      if account_id === :export.to_s
+        puts "\e[91;1mNeed to specify an account id\e[0m\n\n"
+        Dir.rmdir(TMP_DIR)
+        return
+      end
+
+      uri = URI("#{@killbill_url}/#{KILLBILL_API_VERSION}/kb/export/#{account_id}")
+      export_file = TMP_DIR + File::SEPARATOR + 'kbdump'
+
+      request = Net::HTTP::Get.new(uri.request_uri)
```

Comment:
Could you use the Ruby client instead? See https://github.com/killbill/killbill-client-ruby.

Note: it doesn't look like the export endpoint is implemented yet in the client, so it would need to be added first.